api/v1/component_types.go

```diff
@@ -50,6 +51,9 @@ type ComponentSpec struct {
 	ForEach *ForEach `json:"forEach,omitempty"`
 }
 
+// +kubebuilder:validation:Type=object
+type ComponentSpecObject ComponentSpec
+
```

Comment:
Have to declare a new type indirection because kubebuilder didn't process using the same type  `ComponentSpec`